apps/api/src/app/environments-v1/usecases/construct-framework-workflow/construct-framework-workflow.usecase.ts

```diff
@@ -246,7 +253,8 @@ export class ConstructFrameworkWorkflow {
       this.logger.error({ err: error }, 'Failed to evaluate skip rule', LOG_CONTEXT);
     }
 
-    return result;
+    // The Step Conditions in the Dashboard control the step execution, that's why we need to invert the result.
+    return workflowOrigin === WorkflowOriginEnum.NOVU_CLOUD ? !result : result;
```

Comment:
here is where the `origin` is used; I think that comment explains this part

Reply:
please check comment https://github.com/novuhq/novu/pull/7594#discussion_r1930650956